x-pack/plugins/lens/public/visualizations/datatable/components/cell_value.tsx

```diff
@@ -52,22 +59,18 @@ export const createGridCell = (
             if (colorMode === 'cell' && color) {
               style.color = getContrastColor(color, IS_DARK_THEME);
             }
-            setCellProps({
-              style,
-            });
+            setCellProps({ style });
           }
         }
+        if (!colorMode || colorMode === 'none') setCellProps(CELL_RESET_STYLE);
       }
       // make sure to clean it up when something change
       // this avoids cell's styling to stick forever
       return () => {
-        if (minMaxByColumnId?.[originalId]) {
-          setCellProps({
-            style: {
-              backgroundColor: undefined,
-              color: undefined,
-            },
-          });
+        if (!colorMode || colorMode === 'none') {
+          if (minMaxByColumnId?.[originalId]) {
+            setCellProps(CELL_RESET_STYLE);
+          }
```

Comment:
I found a fix to this that just checks `isExpanded` here and prevent unnecessarily calling `setCellProps`.

```suggestion
        if ((!colorMode || colorMode === 'none') && minMaxByColumnId?.[originalId] && !isExpanded) {
          setCellProps(CELL_RESET_STYLE);
        }
```